apm-sniffer/apm-sdk-plugin/logger-plugin/src/main/java/org/apache/skywalking/apm/plugin/logger/ContextConfig.java

```diff
@@ -0,0 +1,233 @@
+                log4j2.setName("log4j2");
+                configs.add(log4j2);
+            }
+            return configs;
+        }
+
+        private static LoggerConfig fillPackageAndLevel(String packages, String level) {
+            LoggerConfig loggerConfig = null;
+            if (packages != null || level != null) {
+                loggerConfig = new LoggerConfig();
+                if (packages != null) {
+                    loggerConfig.setPackages(splitPackages(packages));
+                }
+                if (level != null) {
+                    loggerConfig.setLevel(LogLevel.valueOf(level.toUpperCase()));
+                }
+            }
+            return loggerConfig;
+        }
+
+        private static List<String> splitPackages(String packages) {
+            return Arrays.asList(packages.split(","));
+        }
+
+        /**
+         * @param src instance which need to fill
+         * @return object that has been filled with default values
+         */
+        private static LoggerConfig fillLoggerConfig(LoggerConfig src) {
+            if (src == null) {
+                return null;
+            }
+            if (src.getLevel() == null) {
+                src.setLevel(LogLevel.ERROR);
+            }
+            if (src.getPackages() == null) {
+                List<String> packages = new ArrayList<>();
+                packages.add("*");
+                src.setPackages(packages);
+            }
+            return src;
+        }
+    }
+
+    @Getter
+    @Setter
+    @AllArgsConstructor
+    @NoArgsConstructor
+    static class LoggerConfig implements Cloneable {
+        private String name;
+        private List<String> packages;
+        private LogLevel level;
+
+        /**
+         * Encapsulate the obtained log information into a map
+         *
+         * @param loggerName the name of log system,eg:logback
+         * @param level      which level of log need to recorder
+         * @param args       the params of log
+         * @return a message map
+         */
+        public Map<String, String> toMessageMap(String loggerName, String level, Object... args) {
+            Map<String, String> messageMap = new HashMap<>();
+            messageMap.put("log.kind", loggerName);
+            messageMap.put("event", level);
+            int paramStart = 2;
+            // like `warn(String format, Object arg)`
+            if (args[0] instanceof String) {
+                paramStart = 1;
+                messageMap.put("message", args[0].toString());
+            } else {
+                messageMap.put("message", args[1].toString());
+            }
+            //build log
+            for (int i = paramStart; i < args.length; i++) {
+                String key = "param." + (i - paramStart + 1);
+                messageMap.put(key, args[i].toString());
+            }
+            return messageMap;
+        }
+
+        @Override
+        protected Object clone() throws CloneNotSupportedException {
```

Comment:
Doesn't need to be clonable (nothing clones it).

Reply:
Ok, I will fix it.